pytorch_lightning/trainer/distrib_parts.py

```diff
@@ -540,6 +551,57 @@ def dp_train(self, model):
 
         self.run_pretrain_routine(model)
 
+    def horovod_train(self, model):
+        # Horovod: initialize library
+        hvd.init()
+
+        if torch.cuda.is_available() and self.on_gpu:
+            # Horovod: pin GPU to local rank
+            torch.cuda.set_device(hvd.local_rank())
+            model.cuda(hvd.local_rank())
+
+        self.progress_bar_refresh_rate = self.progress_bar_refresh_rate if hvd.rank() == 0 else 0
+
+        # CHOOSE OPTIMIZER
+        # allow for lr schedulers as well
+        self.optimizers, self.lr_schedulers, self.optimizer_frequencies = self.init_optimizers(model)
+
+        # Horovod: scale the learning rate by the number of workers to account for
+        # increased total batch size
+        for optimizer in self.optimizers:
+            for param_group in optimizer.param_groups:
+                param_group['lr'] *= hvd.size()
+
+        if self.use_amp:
+            # An example
+            model, optimizers = model.configure_apex(amp, model, self.optimizers, self.amp_level)
+            self.optimizers = optimizers
+
+        # Horovod: broadcast parameters & optimizer state to ensure consistent initialization
+        hvd.broadcast_parameters(model.state_dict(), root_rank=0)
+        for optimizer in self.optimizers:
+            hvd.broadcast_optimizer_state(optimizer, root_rank=0)
+
+        # Horovod: wrap optimizers to perform gradient aggregation via allreduce
+        self.optimizers = [
+            hvd.DistributedOptimizer(optimizer, named_parameters=model.named_parameters())
+            for optimizer in self.optimizers
+        ]
+
+        self.proc_rank = hvd.rank()
+        set_proc_rank(self.proc_rank)
+
+        if hvd.rank() != 0:
+            self.logger = None
```

Comment:
is this needed, loggers shall have already `rank_ero_only`
in such case you #1408 and setting global rank
https://github.com/PyTorchLightning/pytorch-lightning/blob/a22a8142ac65668781a6e6f76d3c4e55ea7c249a/pytorch_lightning/trainer/distrib_parts.py#L494

Reply:
The errors come from a race condition where different ranks will attempt to `mkdir` the same directory, leading to an exception being raised on one of the workers.  For example, this can happen when creating a [SummaryWriter](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/loggers/tensorboard.py#L99), which is why in Horovod we only do so on rank 0.

Reply:
in lightning we already handle setting loggers, etc only to rank=0 btw

Reply:
I see.  I updated to set the logger ranks to `hvd.rank()` instead of deleting them outside of rank 0.  Let me know if that makes more sense.